### PR TITLE
Jean S Ariza

### PR DESCRIPTION
**Pull request by jesuarva**
## JS1
- [x] Objects Task
- [x] Stretch Objects Task
- [x] Arrays Task
- [x] Callbacks Task
- [x] Stetch CallbacksTask
- [x] Stretch Task -> ES6 syntaxs

## Good Habits
- [x] Pull Request title is your first and last name (add any notes in the comment field)
- [x] Code has been committed regularly
- [x] Commit comments are informative
- [x] Code is organized in a logical way

### CODE REVIEW

File: assignments/arrays.js [hunk 1]
```diff
@@ -64,33 +67,58 @@ let inventory = [{"id":1,"car_make":"Lincoln","car_model":"Navigator","car_year"
...
 // The marketing team wants the car models listed alphabetically on the website. Sort all the car model names into alphabetical order and log the results in the console
 let carModels = [];
-console.log();
+
+for (let i = 0; i<length; ++i) {
+    // In order to correctly 'sort' the carModels it is neccesary to UpperCase the 'first-letter' of the 'car_model'
+    carModels.push(inventory[i].car_model.charAt(0).toUpperCase()+inventory[i].car_model.slice(1));
+}
+carModels.sort();
+console.log(carModels);
```
Comment (nphillips78) [inline assignments/arrays.js:89]: Good use of slice!

File: assignments/arrays.js [hunk 2]
```diff
@@ -64,33 +67,58 @@ let inventory = [{"id":1,"car_make":"Lincoln","car_model":"Navigator","car_year"
...
+oldCars.forEach( (car) => console.log(car) );
 
 // ==== Challenge 6 ====
 // A buyer is interested in seeing only BMW and Audi cars within the inventory.  Return an array that only contains BMW and Audi cars.  Once you have populated the BMWAndAudi array, use JSON.stringify() to show the results of the array in the console.
 let BMWAndAudi =[];
-console.log();
+for ( let car of inventory ) {
+    ( car.car_make === "BMW" || car.car_make === "Audi" ) ? BMWAndAudi.push(car) : "";
+}
+console.log(BMWAndAudi);
```
Comment (nphillips78) [inline assignments/arrays.js:122]: Good work!

File: assignments/objects.js [hunk 3]
```diff
@@ -19,46 +21,119 @@ let example = {
...
+interns["Gannie"] = {
+  "email": "gmartinson3@illinois.edu",
+  "id": "4",
+  "gender": "M"
+}//?
+// 5,adaine5@samsung.com,Antonietta,F
+interns["Antonietta"] = {
+  "email": "adaine5@samsung.com",
+  "id": "5",
+  "gender": "F"
```
Comment (nphillips78) [inline assignments/objects.js:54]: Excellent

File: assignments/objects.js [hunk 4]
```diff
@@ -19,46 +21,119 @@ let example = {
...
+console.log(interns.Kennan.id);
 
 // Keven's email
+console.log(interns.Keven.email);
 
 // Gannie's name
+console.log(Object.keys(interns)[3]);
 
 // Antonietta's Gender
+console.log(interns.Antonietta.gender);
```
Comment (nphillips78) [inline assignments/objects.js:74]: You could simplify this to `console.log(mitzi.name);`

File: assignments/objects.js [hunk 5]
```diff
@@ -19,46 +21,119 @@ let example = {
...
-// ==== Stretch Challenge: Nested Objects and the this keyword ==== 
+// ==== git push: Nested Objects and the this keyword ==== 
 
 // 1. Create a parent object with properties for name and age.  Make the name Susan and the age 70.
+// const Person = function (name, age ) {
+//   this.name = name;
+//   this.age = age;
+// }
+class Person {
+  constructor ( name, age ){
```
Comment (nphillips78) [inline assignments/objects.js:99]: Fantastic! Love that you created a class 😄

File: assignments/objects.js [hunk 6]
```diff
@@ -19,46 +21,119 @@ let example = {
...
+console.log(`Grandchild is ${parent.child.grandchild.name}, and is ${parent.child.grandchild.age} years old`);
 
 // Have the parent speak
+console.log(parent.speak());
 
 // Have the child speak
+console.log(parent.child.speak());
 
 // Have the grandchild speak
+console.log(parent.child.grandchild.speak());
```
Comment (nphillips78) [inline assignments/objects.js:139]: Great work!

**Review by nphillips78 — COMMENTED** [review]
Excellent job, Jean! I liked seeing you use some of the things we haven't even covered yet :smile:

**Comment by jesuarva** [comment]
Hi, I really appreciate your comments. Thank you!

**Pull request closed without merging**